Makefile

```diff
@@ -58,7 +58,6 @@ endif
 
 build: submodule-update
 	@echo "Building..."
-	go mod tidy
```

Comment:
`go mod tidy` will update `go.mod` and `go.sum`. It should be run by the developer to fixed the dependency versions explicitly. Changing import dependency on every build is NOT desired.

Reply:
Wait I thought `go mod tidy` just simply clean up the dependencies for performance sake, not like upgrading to the latest version or something.

Reply:
Not just remove. It will also add missing dependencies.

> It adds any missing module requirements necessary to build the current module’s packages and dependencies

https://go.dev/ref/mod#go-mod-tidy

Reply:
Don't you need the missing dependencies ...? Or you think that it should create bug and developers set the missing dependencies by himself/herself?

Reply:
> Don't you need the missing dependencies ...?
No, that's why it should be removed. Developer should run `go mod tidy` locally and commit the `go.mod` & `go.sum` file with correct dependency and version. This ensures the pipeline can have the same output on the same commit every time (also take advantage to docker cache). The concept is similar for commit `package-lock.json`.